src/galario.cpp

```diff
@@ -677,12 +677,29 @@ inline dcomplex interpolate_core(int const nrow, int const ncol, const dcomplex
     dcomplex const final_add2 = CMPLXADD(term2, term3);
     dcomplex const final_add1 = CMPLXADD(term1, final_add2);
 
-    dcomplex interpolated = CMPLXADD(final_add1, y0);
+    dcomplex phaseInterp = CMPLXADD(final_add1, y0);
 
     if (u < 0.) {
-        interpolated = CMPLXCONJ(interpolated);
+        phaseInterp = CMPLXCONJ(phaseInterp);
     }
 
+    dreal angleInterp = arg(phaseInterp);
+
+    dreal const tr = indv - fl_v;
+    dreal const qr = indu - fl_u;
+
+    dreal const y0r = abs(y0);
```

Comment:
The `abs` and `arg` functions do not exist in the CUDA math library and raise errors when compiled for GPU.
Following the pattern used for `CMPLXMUL`, `CMPLXADD`, etc. you should create macros such as `CMPLXABS` and `CMPLXARG` that define these operations for the CUDA and CPU implementation.
For CUDA there are cuCabsf and cuCabs (single and double precision) and for CPU you can use the simple abs.

If you need help with this I can do it, if you enable me to write on this PR (there should be an option on the right of the PR page (see instructions [here](https://help.github.com/articles/committing-changes-to-a-pull-request-branch-created-from-a-fork/) on how to enable me).

Reply:
> any idea on how we can make the modified algorithm faster?

I don't know, and I'm actually quite surprised the change in the algorithm is producing such a dramatic performance loss. Do you know if there is any specific part of `interpolate_core` (other than the double interpolation) that is bottlenecking the execution?


> A few changes requested to make the code run on the GPU.

I am adapting the code now, and will commit the changes as soon as possible.